File: Level 1/명예의 전당 (1)/solution.js
function solution(k, scores) {
  const answer = [];
  const stack = [];

  scores.forEach((score) => {
    if (stack.length < k) {
      stack.push(score);
    }

    if (score > stack.at(-1)) {
      stack[stack.length - 1] = score;
    }

    answer.push(Math.min(...stack));
    stack.sort((a, b) => b - a);
  });

  return answer;
}

File: Level 1/명예의 전당 (1)/solution.py
import heapq


def solution(k, scores):
    heap = []

    answer = []

    for score in scores:
        if len(heap) < k:
            heapq.heappush(heap, score)
        else:
            if score > heap[0]:
                heapq.heappush(heap, score)
                heapq.heappop(heap)

        answer.append(heap[0])

    return answer
